python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data_builder.py

```diff
@@ -245,7 +245,11 @@ def build_caching_repository_data_from_list(
             if definition.partitions_def is not None:
                 partitions_defs.add(definition.partitions_def)
             unresolved_jobs[definition.name] = definition
-        elif isinstance(definition, AssetChecksDefinition):
+        elif isinstance(definition, AssetChecksDefinition) or (
+            isinstance(definition, AssetsDefinition)
+            and definition.has_check_keys
+            and not definition.has_keys
+        ):
```

Comment:
I believe the `isinstance` on `AssetChecksDefinition` is redundant here

Reply:
Good point